src/api/layers/wms-capabilities.class.js

```diff
@@ -0,0 +1,174 @@
+import { LayerAttribution } from '@/api/layers/AbstractLayer.class'
+import ExternalWMSLayer from '@/api/layers/ExternalWMSLayer.class'
+import ExternalGroupOfLayers from '@/api/layers/ExternalGroupOfLayers.class'
+import { WMSCapabilities as OlWmsCapabilities } from 'ol/format'
+import log from '@/utils/logging'
+
+/** Wrapper around the OpenLayer WMSCapabilities to add more functionalities */
+export default class WmsCapabilities {
+    constructor(content, originUrl) {
+        const parser = new OlWmsCapabilities()
+        Object.assign(this, parser.read(content))
+        this.originUrl = new URL(originUrl)
+    }
+
+    /**
+     * Find recursively in the capabilities the matching layer ID node
+     *
+     * @param {string} layerId Layer ID to search for
+     * @param {OlWmsCapabilities.Capability.Layer} startFrom Layer node to start from
+     * @returns {OlWmsCapabilities.Capability.Layer} Capability layer node
+     */
+    findLayer(layerId, startFrom = null) {
```

Comment:
I'm not sure I'm a fan of outputting a new type of layer specific to this class/parser outside of this file, that is then checked and transformed into what we usually use through `getExternalLayerObject`.

I would maybe prefer that we transform this node directly in a `ExternalLayer` type (or return `null` of nothing was found), but that means that you have to receive the projection and other things here too...

Reply:
This method is used to search the layers inside the capabilities and cannot return an external layer object !

Reply:
Moreover this class old a Capabilities object and it would be correct to have a `findLayer` member returning one of its layer part. Returning an `ExternalLayer` object would not be correct unless it is specified in the name of the method.
Also I added this method because during parsing of the URL we might need just one `ExternalLayer` out of the whole capabilities, returning all capabilities as `ExternalLayer` in this case would be overkill especially if the capabitilities holds several hundreds of layers.